### PR TITLE
[android/CHIPTool] Add missing parts for rendezvous over BLE

### PR DESCRIPTION
**Pull request by Damian-Nordic**
#### Problem
Android CHIPTool has incomplete implementation of rendezvous over BLE.

 #### Summary of Changes
* Pass Bluetooth Gatt events to CHIP BleLayer
* Fix RendezvousSession in case connection object is given in RendezvousParams instead of discriminator
* Call beginConnectBle() in "Rendezvous over BLE" button handler to initiate secure connection to the device

Note that network provisioning part is still missing.

### CODE REVIEW

File: src/controller/java/CHIPDeviceController-JNI.cpp [hunk 1]
```diff
@@ -365,6 +367,101 @@ JNI_METHOD(void, beginSendCommand)(JNIEnv * env, jobject self, jlong deviceContr
...
+    chip::Ble::ChipBleUUID charUUID;
+    chip::System::PacketBuffer * buffer = nullptr;
+
+    VerifyOrExit(JavaBytesToUUID(env, svcId, svcUUID),
+                 ChipLogError(Controller, "handleIndicationReceived() called with invalid service ID"));
+    VerifyOrExit(JavaBytesToUUID(env, charId, charUUID),
+                 ChipLogError(Controller, "handleIndicationReceived() called with invalid characteristic ID"));
+
+    buffer = System::PacketBuffer::NewWithAvailableSize(valueLength);
+    VerifyOrExit(buffer != nullptr, ChipLogError(Controller, "Failed to allocate packet buffer"));
```
Comment (kkasperczyk-no) [inline src/controller/java/CHIPDeviceController-JNI.cpp:401]: Same as above.

File: src/controller/java/CHIPDeviceController-JNI.cpp [hunk 2]
```diff
@@ -423,14 +520,14 @@ void HandleSimpleOperationComplete(ChipDeviceController * deviceController, void
     jobject self   = (jobject) deviceController->AppState;
     CHIP_ERROR err = CHIP_NO_ERROR;
 
-    ChipLogProgress(Controller, "Received response to %s request", (const char *) appReqState);
+    ChipLogProgress(Controller, "Received response to %s request", operation);
```
Comment (kkasperczyk-no) [inline src/controller/java/CHIPDeviceController-JNI.cpp:523]: Shouldn't we make sure before using that "operation" is not null?

File: src/controller/java/CHIPDeviceController-JNI.cpp [hunk 3]
```diff
@@ -365,6 +367,101 @@ JNI_METHOD(void, beginSendCommand)(JNIEnv * env, jobject self, jlong deviceContr
     }
 }
 
+static bool JavaBytesToUUID(JNIEnv * env, jbyteArray value, chip::Ble::ChipBleUUID & uuid)
+{
+    const auto valueBegin  = env->GetByteArrayElements(value, nullptr);
+    const auto valueLength = env->GetArrayLength(value);
+    bool result            = true;
+
+    VerifyOrExit(valueBegin != nullptr && valueLength == sizeof(uuid.bytes), result = false);
```
Comment (kkasperczyk-no) [inline src/controller/java/CHIPDeviceController-JNI.cpp:376]: I think that != nullptr is redundant here.

**Review by kkasperczyk-no — APPROVED** [review]
LGTM

File: src/controller/java/CHIPDeviceController-JNI.cpp (hunk 1 above)
Comment (Damian-Nordic) [reply src/controller/java/CHIPDeviceController-JNI.cpp:401]: right, will remove

File: src/controller/java/CHIPDeviceController-JNI.cpp (hunk 2 above)
Comment (Damian-Nordic) [reply src/controller/java/CHIPDeviceController-JNI.cpp:523]: The function is only called from this module, so I guess it's safe to assume that it's not null

Comment (kkasperczyk-no) [reply src/controller/java/CHIPDeviceController-JNI.cpp:523]: Ok, nevermind then.

File: src/transport/RendezvousSession.cpp [hunk 4]
```diff
@@ -223,7 +223,7 @@ void RendezvousSession::OnRendezvousConnectionOpened()
 
 void RendezvousSession::OnRendezvousConnectionClosed()
 {
-    if (!mParams.HasDiscriminator() && !mParams.HasConnectionObject())
+    if (!mParams.IsController())
```
Comment (LuDuda) [inline src/transport/RendezvousSession.cpp:226]: Good usage of De Morgan's laws 👍

**Review by LuDuda — COMMENTED** [review]
LGTM 👍 

Btw. Have you added support to sending Write Request vs Write Command? I probably missed it.

**Pull request closed without merging**